test/stateFiltersSpec.js

```diff
@@ -17,6 +20,34 @@ describe('isState filter', function() {
     $q.flush();
     expect($parse('"a" | isState')($rootScope)).toBe(false);
   }));
+
+  it('should return true when the current state is passed with matching parameters', inject(function ($parse, $q, $state, $rootScope) {
+    $state.go('a.c', {x: 'foo', y: 'bar'});
+    $q.flush();
+    expect($parse('"a.c" | isState')($rootScope)).toBe(true);
+    expect($parse('"a.c" | isState:{x: "foo", y: "bar"}')($rootScope)).toBe(true);
+    expect($parse('"a.c" | isState:{x: "bar", y: "foo"}')($rootScope)).toBe(false);
+    expect($parse('"a.c" | isState:{x: "bar"}')($rootScope)).toBe(false);
+    expect($parse('"a.c" | isState:{y: "foo"}')($rootScope)).toBe(false);
+    expect($parse('"a.c" | isState:{x: "bar", y: "foo", z: "hello"}')($rootScope)).toBe(false);
+  }));
+
+  it('should work for relative states', inject(function ($parse, $q, $state, $rootScope) {
+    var scope = $rootScope.$new();
+    scope.options = { relative: $state.get('a') };
+
+    $state.go('a.c');
+    $q.flush();
+
+    expect($parse('".c" | isState:undefined:options')(scope)).toBe(true);
```

Comment:
`isState` filter has been updated to have only one optional parameter as an object, e.g. `".c" | isState:{params:{x: "foo", y: "bar"}, options: options}`